src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -915,6 +915,17 @@ private static bool HadLambdaConversionError(DiagnosticBag diagnostics, Analyzed
             RefKind refArg = arguments.RefKind(arg);
             RefKind refParm = parameter.RefKind;
 
+            if (arguments.IsExtensionMethodThisArgument(arg))
+            {
+                Debug.Assert(refArg == RefKind.None);
+                if (refParm == RefKind.Ref || refParm == RefKind.RefReadOnly)
```

Comment:
For the purposes of overload resolution, ref-readonly is treated as none.

---
In reply to: [122016776](https://github.com/dotnet/roslyn/pull/19703#discussion_r122016776) [](ancestors = 122016776)